hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -458,6 +460,36 @@ public void testDropNamespace() throws TException {
         });
   }
 
+  @Test
+  public void testDropTable() throws TException {
+    Namespace namespace = Namespace.of("dbname_drop");
+    TableIdentifier identifier = TableIdentifier.of(namespace, "table");
+    Schema tableSchema =
+        new Schema(Types.StructType.of(required(1, "id", Types.LongType.get())).fields());
+    catalog.createNamespace(namespace, meta);
+    catalog.createTable(identifier, tableSchema);
+    TableOperations ops = catalog.newTableOps(identifier);
+
+    Assert.assertTrue(catalog.dropTable(identifier, true));
```

Comment:
nit: I feel no need to test line 473 to 480

Reply:
I agree, let's remove L473 - 478 and just focus on the actual case where we don't have a metadata file for the given table